gpusim.cpp

```diff
@@ -231,27 +239,30 @@ void GPUSimServer::newConnection()
     auto server = dynamic_cast<QLocalServer*>(sender());
     QLocalSocket* clientConnection = server->nextPendingConnection();
 
-    // HACK:  For some reason QLocalSocket->serverName doesn't work in socket
-    clientConnection->setProperty("dbname", server->serverName());
-
     QObject::connect(clientConnection, &QLocalSocket::disconnected,
                      clientConnection, &QLocalSocket::deleteLater);
     QObject::connect(clientConnection, &QLocalSocket::readyRead, this,
                      &GPUSimServer::incomingSearchRequest);
 }
 
-void GPUSimServer::searchAll(const Fingerprint& query, int results_requested,
-        float similarity_cutoff, vector<char *>&  results_smiles,
+void GPUSimServer::searchDatabases(const Fingerprint& query,
+        int results_requested,
+        float similarity_cutoff, map<string, QString>& dbname_to_key,
+        vector<char *>&  results_smiles,
         vector<char *>& results_ids, vector<float>& results_scores)
 {
     typedef pair<char*, char*> ResultData;
     typedef pair<float, ResultData > SortableResult;
 
     vector<SortableResult> sortable_results;
-    for(auto local_dbname : m_databases.keys()) {
+    for(auto name_key_pair : dbname_to_key) {
+        const auto& local_dbname = name_key_pair.first;
+        const auto& local_key = name_key_pair.second;
```

Comment:
Do we have C++17 yet?
for (auto& [key, value]: myMap) {

Reply:
I don't want to add that requirement for people trying to build this library, already had people contact me trying to build on CentOS6..